src/System.ComponentModel.EventBasedAsync/tests/AsyncOperationFinalizerTests.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading;
+using Xunit;
+
+namespace System.ComponentModel.Tests
+{
+    public class AsyncOperationFinalizerTests : RemoteExecutorTestBase
+    {
+        [Fact]
+        public void Finalizer_GetViaReflection_IsPresentForCompatability()
+        {
+            MethodInfo finalizer = typeof(AsyncOperation).GetMethod("Finalize", BindingFlags.NonPublic | BindingFlags.Instance);
+            Assert.NotNull(finalizer);
+        }
```

Comment:
> They use the following sync context that the finalizer does indeed call back to when we manually invoke it.

But we shouldn't manually invoke it via reflection.  You instead need to ensure that the object is collectable (e.g. ctor it in a helper method to ensure the JIT doesn't artificially extend its lifetime), and then force collection with GC.Collect and wait for finalizers to run with GC.WaitForPendingFinalizers.